explorer/explorer.go

```diff
@@ -0,0 +1,46 @@
+package explorer
+
+import (
+	"errors"
+	"gioui.org/io/event"
+)
+
+var (
+	// ErrUserDecline is returned when the user doesn't select the file.
+	ErrUserDecline = errors.New("user exit the file selector without selecting a file")
+
+	// ErrNotAvailable is return when the current OS isn't supported.
+	ErrNotAvailable = errors.New("current OS not supported")
+)
+
+// ListenEvents must get all the events from Gio, in order to get the GioView. You must
+// include that function where you listen for Gio events.
+//
+// Similar as:
+//
+// select {
+// case e := <-window.Events():
+// 		explorer.ListenEvents(e)
+// 		switch e := e.(type) {
+// 				(( ... your code ...  ))
+// 		}
+// }
+func ListenEvents(event event.Event) {
+	listenEvents(event)
+}
+
+// ReadFile opens the file selector, allowing the user to select a single file.
+// It will open the default folder. Optionally, the extensions to filter
+// the file extension (such as `.jpg`, `.png`).
+//
+// Example: ReadFile(".jpg", ".png") will only accept the selection of files with
+// .jpg or .png extensions.
+func ReadFile(extensions ...string) ([]byte, error) {
```

Comment:
I choose to return the data as `[]byte` instead of `io.ReadCloser` or `path string` because we have some limitations across the OSes:

1. Some OSes doesn't have "folders". Consider WASM/JS for instance, we don't know here the file comes from. The Android have some issues too, since the "path" is something like `content://` and `os.Open` fails sometimes.

2. Wrapping the `[]byte` into some `io.NopCloser(bytes.NewReader(content))` can be misleading. When it's explicit that it returns `[]byte` we know that loading large files can be a issue. However, by wrapping it we hide such problem, but not fix them. 

For know, using `[]byte` seems the right choice.